bot/connector-teams/src/main/kotlin/token/TokenHandler.kt

```diff
@@ -0,0 +1,85 @@
+package fr.vsct.tock.bot.connector.teams.token
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.PropertyNamingStrategy
+import fr.vsct.tock.shared.Level
+import fr.vsct.tock.shared.addJacksonConverter
+import fr.vsct.tock.shared.create
+import fr.vsct.tock.shared.jackson.mapper
+import fr.vsct.tock.shared.longProperty
+import fr.vsct.tock.shared.retrofitBuilderWithTimeoutAndLogger
+import mu.KotlinLogging
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import kotlin.concurrent.fixedRateTimer
+
+/**
+ * Handle the generation and the refresh of the token header
+ * This token is mandatory in request from bot to teams via microsoft-api
+ */
+object TokenHandler {
+
+    private val logger = KotlinLogging.logger {}
+
+    @Volatile
+    var token: String? = null
+    @Volatile
+    private var tokenExpiration: Instant? = null
+
+    private lateinit var appId: String
```

Comment:
appId and password should be Volatile too

Reply:
good point